### PR TITLE
Add scope for node placement and numerical_hint function

### PR DESCRIPTION
**Pull request by samikama**
This PR adds an nvidia_scope for users to be able to fine-tune trt conversion process and a numerical_hint function for future user defined quantization ranges in TRT.

### CODE REVIEW

**Comment by samikama** [comment]
@aaroey can you add @pooyadavoodi to reviewers please. I can't seem to be able to add him

**Comment by aaroey** [comment]
@samikama I was not able to do so either. His id doesn't show up in the reviewers list.

File: tensorflow/contrib/tensorrt/BUILD [hunk 1]
```diff
@@ -176,7 +176,11 @@ py_library(
 
 py_library(
     name = "trt_convert_py",
-    srcs = ["python/trt_convert.py"],
+    srcs = [
+        "python/nvidia_scope.py",
```
Comment (aaroey) [inline tensorflow/contrib/tensorrt/BUILD]: nit: please rename to `nv_scope.py`.

File: tensorflow/contrib/tensorrt/test/base_test.py [hunk 2]
```diff
@@ -119,6 +121,57 @@ def GetParams(self):
         allclose_atol=1.e-03,
         allclose_rtol=1.e-03)
 
+class SimpleScopedGraphDefTest(trt_test.TfTrtIntegrationTestBase):
```
Comment (aaroey) [inline tensorflow/contrib/tensorrt/test/base_test.py]: Please move this to a separate `_test.py` file.

**Review by aaroey — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/tensorrt/python/nvidia_scope.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    precision=None,
+    engine_mode=None,
+    maximum_cached_engines=None,
+    maximum_workspace_size=None,
+):
+  """Include or exclude nodes in optimized segment with given parameters.
+
+  NOTE: this is a hint and will be supported on a best-effort basis.
+  Args:
+    include      : Whether to include the node or not.
```
Comment (pooyadavoodi) [inline tensorflow/contrib/tensorrt/python/nvidia_scope.py]: Looks like you have renamed `include` to `select` in the code but forgot to rename it in the comments?

File: tensorflow/contrib/tensorrt/python/quantization_hints.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+from __future__ import print_function
+
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import constant_op as cop
+from tensorflow.python.framework import dtypes
+from tensorflow.core.framework import attr_value_pb2
+
+
+def numerical_hint(op, max_value, min_value):
+  """Add a const numerical hint node"""
```
Comment (pooyadavoodi) [inline tensorflow/contrib/tensorrt/python/quantization_hints.py]: Could you describe in the comment what exactly this function and the arguments `max_value` and `min_value` do?

File: tensorflow/contrib/tensorrt/test/base_test.py [hunk 5]
```diff
@@ -119,6 +121,57 @@ def GetParams(self):
...
+            name="conv")
+        c1 = constant_op.constant(
+            np.random.randn(input_dims[0], 12, 12, 6), dtype=dtype)
+        p = conv * c1
+        c2 = constant_op.constant(
+            np.random.randn(input_dims[0], 12, 12, 6), dtype=dtype)
+        q = conv / c2
+
+        edge = self.trt_incompatible_op(q)
+        with nv_scope(select=False):
```
Comment (pooyadavoodi) [inline tensorflow/contrib/tensorrt/test/base_test.py]: This is only testing `select=False`.
How about `select=True`?
How about testing other arguments of `nv_scope`?

**Comment by tensorflowbutler** [comment]
It has been 49 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**